### DISCUSSION TITLE
Add a Dishwasher device

### DISCUSSION THREAD

**Discussion by normanr** [andrei-tatar/node-red-contrib-smartnora#167] [Ideas]
I've recently added a smart plug to my old dishwasher (so that I can use it to do delayed start). Google Assistant has a Dishwasher device type and it would be nice to use it in SmartNora (instead of the generic outlet type).

**Comment by andrei-tatar** [comment]
@normanr sure, sounds useful. Does the start/stop trait bring any value to you?
https://developers.google.com/assistant/smarthome/traits/startstop

**Reply by normanr** [reply]
The assistant API docs note that "[OnOff](https://developers.google.com/assistant/smarthome/traits/onoff) is needed only if the dishwasher has a separate power control from start/stop semantics; otherwise, use [StartStop](https://developers.google.com/assistant/smarthome/traits/startstop)". I don't have any value for OnOff _and_ StartStop, either would be fine (the plug switch I'm using at the moment is just a "regular" Linkind smart plug).

**Reply by normanr** [reply]
Maybe like `media` can be any one of a list of devices: You could have an `appliance` that could be a `dishwasher`, `dryer`, `mop`, `mower`, `vacum`, `washer` (all things with `RunCycle` trait). Things like `blender`, `coffee maker`, `dehydrator`, `oven` (and the existing `sousvide` device), could be bundled into things with a `Cook` trait.

**Comment by normanr** [comment]
I'd also like `oven` to be added. I want to add my Sauna and it seems that an oven device might the closest way to model it (Google doesn't have a Sauna device in its model). I can probably use the `sousvide` device for now as it has the right sort of traits.
